Feature/maksim

BalashovMaksim opened the pull request.

BalashovMaksim committed:
Добавил документацию к BackendApplication

BalashovMaksim committed:
Добавил:
-конфигурацию liquibase,
-файл changeset.xml
-несколько новых зависимостей
-.env файл
-класс APIConfig.java для настройки зависмости dotenv

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+    <changeSet author="Maksim Balashov" id="create-person-table">
+        <sql>
+            CREATE TABLE person (
+            id BIGSERIAL PRIMARY KEY,
```
id UUID PRIMARY KEY здесь и во всех других таблицах

BalashovMaksim replied:
окау босс

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+            CREATE TABLE person (
+            id BIGSERIAL PRIMARY KEY,
+            first_name VARCHAR(20) NOT NULL,
+            last_name VARCHAR(25) NOT NULL,
```
Давай уберём нот_нулл, пусть будет возможности удалить дату рождения, если чел не хочет к себе лишнего внимания

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+            first_name VARCHAR(20) NOT NULL,
+            last_name VARCHAR(25) NOT NULL,
+            birthday DATE NOT NULL,
+            transplantation_date DATE
```
Также давай добавим сюда поля:
is_activated BOOL DEFAULT FALSE
is_deleted BOOL DEFAULT FALSE
Первое нужно для обозначения активированный пользователя, который не просто спарсился из ВК, но и прошёл дополнительную регистрацию
Второе нужно для так называемого "мягкого удаления" - значимые сущности часто в серьёзных проектах не удаляются полностью, а помечаются как is_deleted = true, то есть она не используется, но в таблице продолжает лежать. Так мы сможем потом восстанавливать пользователей если надо и вести аудит удалённых пользователей, например

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+        <sql>
+            CREATE TABLE person_organ (
+            id BIGSERIAL PRIMARY KEY,
+            person_id BIGINT NOT NULL,
```
Ссылки все тоже переделай на UUID

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+            id BIGSERIAL PRIMARY KEY,
+            santa_id BIGINT NOT NULL,
+            recipient_id BIGINT NOT NULL,
+            file_id BIGINT NOT NULL,
```
Может быть нулл, если его ещё не прикрепили

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+            santa_id BIGINT NOT NULL,
+            recipient_id BIGINT NOT NULL,
+            file_id BIGINT NOT NULL,
+            status VARCHAR(20),
```
Не может быть нулл, можно даже прописать NOT NULL DEFAULT 'CHOOSING'

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+            recipient_id BIGINT NOT NULL,
+            file_id BIGINT NOT NULL,
+            status VARCHAR(20),
+            comment VARCHAR(255),
```
255 может не хватить, если там будет какое-то объяснение. Давай сделаем TEXT

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+                http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.0.xsd">
+
+    <changeSet author="Maksim Balashov" id="create-person-table">
+        <sql>
```
В идеале ещё добавлять precondition

Типа
<precondition>
    <not>
        <tableExists tableName = 'person'/>
    </not>
</precondition>

Как-то так

yaroslav-sapozhnikov commented on src/main/resources/db/changelog/changeset/changeset.xml:
```diff
@@ -0,0 +1,72 @@
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.0.xsd">
+
+    <changeSet author="Maksim Balashov" id="create-person-table">
```
И ещё думаю лучше айдишники указывать в виде даты, типа "20240111-01" - то есть дата плюс номер сета за эту дату. Может потом быть ещё 02, 03 и тд. Конечно это не обязательно, но мне кажется так удобнее

BalashovMaksim committed:
Добавил:
-конфигурацию liquibase,
-файл changeset.xml
-несколько новых зависимостей
-.env файл
-класс APIConfig.java для настройки зависмости dotenv

BalashovMaksim committed:
Добавил:
-конфигурацию liquibase,
-файл changeset.xml
-несколько новых зависимостей
-.env файл
-класс APIConfig.java для настройки зависмости dotenv
